faiss/gpu/GpuIndexIVF.cu

```diff
@@ -426,18 +431,40 @@ void GpuIndexIVF::trainQuantizer_(idx_t n, const float* x) {
         return;
     }
 
-    if (this->verbose) {
-        printf("Training IVF quantizer on %ld vectors in %dD\n", n, d);
+    printf("Training IVF quantizer on %ld vectors in %dD\n", n, d);
+
+    if (config_.use_raft) {
+        printf("Using raft to train quantizer for %d vectors\n", n);
+        const raft::handle_t& raft_handle =
+                resources_->getRaftHandleCurrentDevice();
+
+        raft::neighbors::ivf_flat::index_params raft_idx_params;
+        raft_idx_params.n_lists = nlist;
+        raft_idx_params.metric = raft::distance::DistanceType::L2Expanded;
+        raft_idx_params.add_data_on_build = false;
+        raft_idx_params.kmeans_trainset_fraction = 1.0;
+        raft_idx_params.kmeans_n_iters = 100;
+
+        auto raft_index = raft::neighbors::ivf_flat::build(
+                raft_handle, raft_idx_params, x, n, (idx_t)d);
+
+        raft_handle.sync_stream();
+
+        // TODO: Validate this is all we need to do
+        quantizer->reset();
```

Comment:
FYI, as of this commit:

https://github.com/facebookresearch/faiss/commit/16d5ec755f3f7615668f160a4f23fcbf3dd6ac7b

both CPU and GPU indices can be used as a coarse quantizer (so it follows the CPU implementation better). For RAFT usage it seems ok to simply not allow pluggable CPU coarse quantizers, but if there's a way to pass in pre-assigned IVF clusters to look at for RAFT IVF indices, that would be nice.